p2p.go

```diff
@@ -288,9 +299,23 @@ func NewHost(ctx context.Context, options ...Option) (*Host, error) {
 		libp2p.ListenAddrStrings(fmt.Sprintf("/ip4/%s/tcp/%d", ip, cfg.Port)),
 		libp2p.AddrsFactory(func(addrs []multiaddr.Multiaddr) []multiaddr.Multiaddr {
 			if extMultiAddr != nil {
-				return append(addrs, extMultiAddr)
+				addrs = append(addrs, extMultiAddr)
+			}
+			if cfg.EnablePrivateIP {
+				return addrs
+			}
+			// only public addr is advertised to the p2p network.
+			ret := []multiaddr.Multiaddr{}
+			for _, addr := range addrs {
+				if manet.IsPublicAddr(addr) {
+					ret = append(ret, addr)
+				}
+			}
+			if len(ret) == 0 {
+				// if no public addr is found, private ip is returned
+				return addrs
```

Comment:
if no public ip found, returning private ips may not be appropriate. what's your concern of returning empty array?